src/callbacks_step/amr_dg1d.jl

```diff
@@ -76,6 +76,112 @@ function refine!(u_ode::AbstractVector, adaptor, mesh::TreeMesh{1},
     return nothing
 end
 
+function refine!(u_ode::AbstractVector, adaptor, mesh::TreeMesh{1},
```

Comment:
Maybe one could do it with a keyword argument `cache_parabolic=nothing` and then check in the body whether `cache_parabolic != nothing`?

Reply:
The disadvantage with that approach is that it doesn't stack very well, and it requires the purely hyperbolic person to consider also code for parabolic stuff. This is why we usually try to avoid such constructs and instead try to call one function from the other.

Is there a specific reason why you don't want to implement this based on dispatch, if I may ask?

Reply:
_"Is there a specific reason why you don't want to implement this based on dispatch, if I may ask?"_ 

Not really sure how to do this, since there is this additional datastructure `cache_parabolic` coming in, not a different `cache` of, say, type `CacheParabolic`.

Reply:
But the `cache_parabolic` is only present when this is called for a hyperbolic-parabolic semi, right? So couldn't you implement it as something like
```julia
function refine!(bla, blubb, cache)
    ...
end

function refine!(bla, blubb, cache, cache_parabolic)
    refine!(bla, blubb, cache)
    # whatever is needed for cache_parabolic
end
```
?

Reply:
Yeah that seems to work, please have a look.

Reply:
Yes, looks much better imho! As far as I can tell, this implementation has as the only objective overhead the duplicate call to `local_leaf_cells` (which would be ok for me).

Would something similar (deduplication by stacked calls) for the implementation in the file `amr.jl`?

Reply:
Hm, without adding `if` clauses or keywords to the hyperbolic part I do not see an immediate way that avoids additional datastructures and/or recomputations (one would at least need to preserve `elements_to_refine` and `elements_to_remove`.

Reply:
OK, then let's keep it as it is for now!